src/main/kotlin/com/sourcegraph/cody/edit/widget/LensSpinner.kt

```diff
@@ -1,61 +1,15 @@
 package com.sourcegraph.cody.edit.widget
 
-import com.intellij.openapi.application.runInEdt
 import java.awt.FontMetrics
 import java.awt.Graphics2D
-import java.awt.geom.AffineTransform
 import javax.swing.Icon
-import javax.swing.Timer
 
 class LensSpinner(group: LensWidgetGroup, private val icon: Icon) : LensWidget(group) {
-  private var rotationDegrees = 0f
-  private val animationDelay = 50 // Milliseconds between frames
-
-  private val timer =
```

Comment:
why is it redundant now? 🤔

Reply:
I think the issue here was that the test was failing sporadically because the spinner wasn't getting disposed. But @pkukielka found a way to do it without that code, using an icon with rotation information encoded in it.

Reply:
That is correct :)

The real change which enabled it was:
```
- icon.paintIcon(null, g, x.toInt(), y.toInt())
+ icon.paintIcon(parentGroup.editor.contentComponent, g, x.toInt(), y.toInt())
```

Proper definition of component from which icon painting should get it properties fixed the default animation and allowed to remove all the custom code.

Reply:
awesome